### PR TITLE
Doc treewide push

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Treewide changes often result in regressions (build error and warnings) that slip through the PR run but get caught by the push one, and then results in false postive on other PRs until the original change is either fixed or reverted.

This can be avoided by doing a manual push run in zephyr-testing, add a note in the documentation on how to do it.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Hey, thinking about suggesting this as an approach to mitigate big patches CI breakages issues. I'm thinking ideally we could add a tag to the PR or something and make it trigger the push run without having to use the testing repo, but this may close the gap in the meantime, what do you think? @zephyrproject-rtos/release

**Review by nashif — CHANGES_REQUESTED** [review]
I do not think this should be documented or available to the occasional contributor or developer. We are trying to control and manage CI resources being used and opening the gates for such changes somewhere else that is not monitored is just to going to be a big mess. zephyr-testing should remain 'invite only' and if we need to improve CI, we should do it on the main tree.

File: doc/contribute/guidelines.rst
Comment (fabiobaltieri) [reply doc/contribute/guidelines.rst:1040]: Yeah fair enough, sorry I failed to make it clear in the proposed text but the target I had in mind for that paragraph was the release team, as in "before merging a treewide consider pushing it in testing" or "ask someone for release to run it for you". At least that's something I do regularly for my patches and I pushed some for others as well, and ultimately that saved plenty of CI breakages.

But yeah ideally this would trigger automatically with the treewide tag or something more refined, was half thinking on how to implement that but I guess it probably needs some refactoring  of the workflow, or maybe some logic to switch from the PR to the push run.

Comment (fabiobaltieri) [reply doc/contribute/guidelines.rst:1040]: But at least it seems like we agree that we needs something there, every treewide ends up breaking the push CI and then we start screaming "revert" and the author has to fix it in hurry etc etc. Not ideal.

Comment (stephanosio) [reply doc/contribute/guidelines.rst:1040]: Simple label-based triggering of a push-equivalent run can be prone to abuse because anyone with just triage permission level (i.e. all collaborators and maintainers) can add the label and trigger the _expensive_ push-equivalent run on any PR.

I will give this some more thought and see if there is a _safe_ way of implementing this using the current GitHub feature limitations.

**Pull request closed without merging**
